### PR TITLE
Fix case that was causing us to not offer completion after a case-pattern-label in a switch.

### LINKED ISSUE #15348: Missing completion list after switch-based pattern matching
Paste C#:

```C#
class C
{
    void M()
    {
        object o = 1;
        switch(o)
        {
            case int i:
                $$
                break;
        }

        int i = 7;
        switch(i)
        {
            case 1:
                $$
                break;
        }
    }
}
```

No completion list shows at the first $$ (pattern matching), but the completion list does show at the second $$ (regular switch).

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes #15348

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
retest windows_build_correctness_prtest please

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 

**Customer scenario**

user cannot use completion after a "pattern case label" (added when we added pattern matching).

**Bugs this fixes:**

Fixes #15348

**Workarounds, if any**

none.

**Risk**

Low.

**Performance impact**

low.

**Is this a regression from a previous update?**

no.  

**Root cause analysis:**

It was not missed.  We found it when we tested pattern matching.  But the bug has never been fixed.  it's driving me crazy, so i'm just fixing it :)

**How was the bug found?**

Found during pattern matching test pass.

**Comment by CyrusNajmabadi** [comment]
This was also code reviewed in https://github.com/dotnet/roslyn/pull/19494

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

File: src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb [hunk 1]
```diff
@@ -3311,6 +3311,31 @@ class C
...
+            case int i:
+                $$
+                break;
+        }
+    }
+}
+                              </Document>)
+
+                    state.SendTypeChars("this")
+                    Await state.AssertSelectedCompletionItem(displayText:="this", isHardSelected:=True)
```
Comment (Pilchie) [inline src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb]: I'd love to see a test at the keyword recommender and symbol provider as well

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz

**Pull request merged**